zend_abstract_interface/symbols/call.c

```diff
@@ -2,14 +2,89 @@
 
 #include <ctype.h>
 #include <sandbox/sandbox.h>
-#include <zai_assert/zai_assert.h>
 
-static inline void zai_symbol_call_argv(zend_fcall_info *fci, uint32_t argc, va_list *args ZAI_TSRMLS_DC) {
+#if PHP_VERSION_ID >= 70300
```

Comment:
The motivation for the use of an arena comes from Zend - it uses an arena. Because of implementation details however, we are not able to use it. First, Zend uses the compilers arena and updates the function table. We don't want to do that, so we use an arena we destroy after the call. Second, we extract added benefit - making the allocation of arguments quicker, the runtime cache invisible, and leaving the call faster.

Reply:
I added a comment.